python/ray/worker.py

```diff
@@ -544,32 +543,20 @@ def get_object(self, object_ids):
         assert len(final_results) == len(object_ids)
         return final_results
 
-    def submit_task(self,
-                    function_descriptor,
-                    args,
-                    actor_id=None,
-                    actor_handle_id=None,
-                    actor_counter=0,
-                    actor_creation_id=None,
-                    actor_creation_dummy_object_id=None,
-                    max_actor_reconstructions=0,
-                    execution_dependencies=None,
-                    new_actor_handles=None,
-                    num_return_vals=None,
-                    resources=None,
-                    placement_resources=None,
-                    driver_id=None):
+    def submit_task(self, function_descriptor_list, function_signature,
+                    func_args, func_kwargs, **kwargs):
```

Comment:
I don't like this change. I think it makes readability worse and makes it harder to refactor. If the keyword arguments are a performance issue, why not get rid of the Python function entirely and just call the Cython version?

Reply:
I agree; I did it this way because I wanted to keep the task_index in the worker. I'll try to put it into the RayletClient and see where that leaves us.